tests/nested/core/core-snap-refresh-on-core/task.yaml

```diff
@@ -18,6 +18,11 @@ execute: |
     INITIAL_REV="$(get_nested_core_revision_for_channel "${CORE_CHANNEL}")"
     NEW_REV="$(get_nested_core_revision_for_channel "${NEW_CORE_CHANNEL}")"
 
+    if [ "$INITIAL_REV" = "$NEW_REV" ]; then
+        echo "Initial and target revisions are the same, skipping..."
```

Comment:
agreed nice to have, but let's do this in a followup it would be great to land this as-is as the PR has gotten really big and difficult to review as-is